cpp/translate_chainer.cc

```diff
@@ -149,19 +149,30 @@ void InitialCall(const std::multimap<std::string, std::string> &cgi_args) {
 
 
 // A standard call should provide "index", "language_code", "category" and "translation".
-void RegularCall(const std::multimap<std::string, std::string> &cgi_args) {
+void Insert(const std::multimap<std::string, std::string> &cgi_args) {
     const std::string language_code(GetCGIParameterOrDie(cgi_args, "language_code"));
     const std::string translation(GetCGIParameterOrDie(cgi_args, "translation"));
     const std::string index(GetCGIParameterOrDie(cgi_args, "index"));
     const std::string category(GetCGIParameterOrDie(cgi_args, "category"));
 
+    if (translation.empty())
+        return;
+    
     const std::string INSERT_COMMAND("/usr/local/bin/translation_db_tool insert " + index + " " + language_code
                                      + " " + category + " " + translation);
     std::string output;
     if (not ExecUtil::ExecSubcommandAndCaptureStdout(INSERT_COMMAND, &output))
         Error("failed to execute \"" + INSERT_COMMAND + "\" or it returned a non-zero exit code!");
+}
 
-    ParseTranslationsDbToolOutputAndGenerateNewDisplay(output, language_code);
+
+void EmitHeader() {
+    std::cout << "Content-Type: text/html; charset=utf-8\r\n\r\n";
+}
+
+void EmitRedirectHeader(const std::string language_code) {
+    std::cout << "Status: 302 Found\r\n";
+    std::cout << "Location: /cgi-bin/translate_chainer?language_code=" << language_code << "\r\n\r\n";
 }
```

Comment:
I assume we need to emit the same headers as in the other case or you need to call that function, too.